core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/protocol/restxml/XmlPayloadMarshaller.java

```diff
@@ -0,0 +1,255 @@
+                             ValueToStringConverter.ValueToString<SdkBytes> converter) {
+            context.xmlGenerator().xmlWriter()
+                   .startElement(paramName)
+                   .value(val.asByteBuffer())
+                   .endElement();
+        }
+    };
+
+    public static final XmlMarshaller<SdkPojo> SDK_POJO = new BasePayloadMarshaller<SdkPojo>(null) {
+        @Override
+        public void marshall(SdkPojo val, XmlMarshallerContext context, String paramName,
+                             SdkField<SdkPojo> sdkField, ValueToStringConverter.ValueToString<SdkPojo> converter) {
+            context.xmlGenerator().startElement(paramName);
+            context.protocolMarshaller().doMarshall(val);
+            context.xmlGenerator().endElement();
+        }
+    };
+
+    public static final XmlMarshaller<List> LIST = new BasePayloadMarshaller<List>(null) {
+        @Override
+        public void marshall(List list, XmlMarshallerContext context, String paramName,
+                             SdkField<List> sdkField, ValueToStringConverter.ValueToString<List> converter) {
+            ListTrait listTrait = sdkField
+                .getOptionalTrait(ListTrait.class)
+                .orElseThrow(() -> new IllegalStateException("SdkField of list type is missing List trait"));
+
+            if (!listTrait.flattened()) {
+                context.xmlGenerator().startElement(paramName);
+            }
+
+            final SdkField memberField = listTrait.memberFieldInfo();
+            final LocationTrait memberLocationTrait = (LocationTrait) memberField.getTrait(LocationTrait.class);
+            final String memberLocationName = listMemberLocationName(listTrait, paramName);
+
+            for (Object listMember : list) {
+                context.marshall(memberLocationTrait.location(), listMember, memberLocationName, memberField);
+            }
+
+            if (!listTrait.flattened()) {
+                context.xmlGenerator().endElement();
+            }
+        }
+
+        private String listMemberLocationName(ListTrait listTrait, String listLocationName) {
+            String locationName = listTrait.memberLocationName();
+
+            if (locationName == null) {
+                locationName = listTrait.flattened() ? listLocationName : "member";
+            }
+
+            return locationName;
+        }
+
+        @Override
+        protected boolean shouldEmit(List list, String paramName) {
+            return super.shouldEmit(list, paramName) &&
+                   (!list.isEmpty() || !(list instanceof SdkAutoConstructList));
+        }
+    };
+
+    public static final XmlMarshaller<Map> MAP = new BasePayloadMarshaller<Map>(null) {
+
+        @Override
+        public void marshall(Map map, XmlMarshallerContext context, String paramName,
+                             SdkField<Map> sdkField, ValueToStringConverter.ValueToString<Map> converter) {
+            context.xmlGenerator().startElement(paramName);
```

Comment:
I think we have a test in the protocol suite that tests for flattened maps. I don't see any actual usage of it in the rest-xml services though. Maybe just make a note that we explicitly don't care about the flattened trait for maps.